### PR TITLE
pyqasm validate via CLI

### PR DESCRIPTION
**Pull request by ryanhill1**
## Summary of changes

Added `pyqasm.cli` module with `typer` integration to enable using pyqasm as a command-line tool. For example:

```bash
$ pip install 'pyqasm[cli]'
$ pyqasm --help
Usage: pyqasm [OPTIONS] COMMAND [ARGS]...
$ pyqasm --version
pyqasm/0.1.0a1
$ pyqasm validate tests/cli/resources
tests/cli/resources/invalid1.qasm: error: Index 2 out of range for register of size 1 in qubit [validation]
Found errors in 1 file (checked 3 source files)
$ pyqasm validate tests/cli/resources --skip tests/cli/resources/invalid1.qasm
Success: no issues found in 2 source files
```

### CODE REVIEW

File: src/pyqasm/cli/validate.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+            line_number += 1
+            if 5 <= line_number <= 30 and skip_tag in line:
+                return True
+            if line_number > 30:
+                break
+
+        return False
+
+    def validate_qasm_file(file_path: str) -> None:
+        with open(file_path, "r", encoding="ISO-8859-1") as f:
```
Comment (TheGupta2012) [inline src/pyqasm/cli/validate.py:66]: any particular reason for not choosing the `utf-8` encoding here?

File: src/pyqasm/cli/validate.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+            content = f.read()
+
+        if should_skip(file_path, content):
+            return
+
+        try:
+            module = load(file_path)
+            module.validate()
+        except Exception as err:  # pylint: disable=broad-exception-caught
```
Comment (TheGupta2012) [inline src/pyqasm/cli/validate.py:75]: I think it is better to catch the `pyqasm.exceptions.ValidationError` exception here

File: src/pyqasm/cli/validate.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    def should_skip(filepath: str, content: str) -> bool:
+        if filepath in skip_files:
+            return True
+
+        skip_tag = "// pyqasm: ignore"
+        line_number = 0
+
+        for line in content.splitlines():
+            line_number += 1
+            if 5 <= line_number <= 30 and skip_tag in line:
```
Comment (TheGupta2012) [inline src/pyqasm/cli/validate.py:58]: How are `5` and `30` chosen here?

File: src/pyqasm/cli/validate.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+                return True
+            if line_number > 30:
```
Comment (TheGupta2012) [inline src/pyqasm/cli/validate.py:60]: Here too

File: src/pyqasm/cli/validate.py (hunk 1 above)
Comment (ryanhill1) [reply src/pyqasm/cli/validate.py:66]: No good reason, copied most of this code from the qbraid-cli headers check. But can definitely change it to `utf-8`. That would probably make more sense.

File: src/pyqasm/cli/validate.py (hunk 2 above)
Comment (ryanhill1) [reply src/pyqasm/cli/validate.py:75]: Aren't there other unroll exceptions or parsing exceptions that are also possible? Want to make sure we catch all exceptions, cause if we don't the command will fail with error without providing any helpful info. Can definitely add more granularity in catching these exceptions though

File: src/pyqasm/cli/validate.py (hunk 3 above)
Comment (ryanhill1) [reply src/pyqasm/cli/validate.py:58]: No good reason, carried over from qbraid-cli headers check. Basically defines the bounds of where we check for for the skip comment. But can be removed, or require that file skip comment is on the first line.

File: src/pyqasm/cli/validate.py (hunk 2 above)
Comment (TheGupta2012) [reply src/pyqasm/cli/validate.py]: Yep, thanks for catching that

**Pull request merged**